element-connector/src/main/java/org/eclipse/californium/elements/util/ClockUtil.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.elements.util;
+
+/**
+ * Clock utility. Provides a {@link #Realtime} handler.
+ * 
+ * Intended to be used for system as android, where {@link System#nanoTime()}
+ * reflects the system uptime without system sleep times. Android devices, which
+ * want to use californium timer's (e.g. auto resumption or deduplication) in
+ * sleep times, must set their handler and provide the proper expired
+ * nanoseconds. This will not wake up any californium timer in time, but it
+ * expires them on then next wake up. Not used for schedule timed of schedule
+ * timer e.g. {@link ExecutorsUtil}.
+ */
+public class ClockUtil {
+
+	/**
+	 * Handler to return the expired realtime in nanoseconds.
+	 */
+	public interface Realtime {
+
+		/**
+		 * Get the system specific expired realtime in nanoseconds.
+		 * 
+		 * @return expired realtime in nanoseconds.
+		 */
+		long nanoRealtime();
+	}
+
+	/**
+	 * Handler for system specific expired realtime in nanoseconds.
+	 */
+	private static volatile Realtime handler;
+
+	/**
+	 * Set handler system specific expired realtime in nanoseconds.
+	 * 
+	 * @param systemHandler system specific expired realtime in nanoseconds.
+	 */
+	public static void setRealtimeHandler(Realtime systemHandler) {
+		handler = systemHandler;
+	}
+
+	/**
+	 * Get expired realtime in nanoseconds.
+	 * 
+	 * If no system specific handler was set before, {@link System#nanoTime()}
+	 * is used.
+	 * 
+	 * @return expired realtime in nanoseconds
+	 */
+	public static long nanoRealtime() {
```

Comment:
I'm not sure if this is better, but intuitively I would have created a default RealTime implementation which returns `System.nanoTime` instead of doing the check `systemHandler !=  nul`l for each call of nanoRealtime.

Reply:
OK.

Reply:
Done.